src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingMapListController.java

```diff
@@ -0,0 +1,205 @@
+import javafx.scene.layout.Pane;
+
+import com.faforever.client.domain.server.MatchmakerQueueInfo;
+import com.faforever.client.fx.NodeController;
+import com.faforever.client.map.MapService;
+import javafx.scene.layout.VBox;
+import javafx.stage.Screen;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+
+public class TeamMatchmakingMapListController extends NodeController<Pane> {
+
+  private static final int TILE_SIZE = 125;
+  private static final int PADDING = 20;
+
+  private final MapService mapService;
+  private final UiService uiService;
+  private final PlayerService playerService;
+  private final FxApplicationThreadExecutor fxApplicationThreadExecutor;
+
+  public Pane root;
+  public FlowPane tilesContainer;
+  public ScrollPane scrollContainer;
+  public VBox loadingPane;
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBrackets;
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBracketsWithDuplicates;
+  private Integer playerBracketIndex = null;
+  @Setter
+  private double maxWidth = 0;
+  @Setter
+  private double maxHeight = 0;
+
+
+  @Override
+  protected void onInitialize() {
+    JavaFxUtil.bindManagedToVisible(loadingPane);
+  }
+
+  @Override
+  public Pane getRoot() {
+    return root;
+  }
+
+
+  public void setQueue(MatchmakerQueueInfo queue) {
+    mapService.getMatchmakerBrackets(queue).subscribe(rawBrackets -> {
+      loadingPane.setVisible(false);
+
+      this.sortedBracketsWithDuplicates = this.getSortedBrackets(rawBrackets);
+
+      this.sortedBrackets = new TreeMap<>(this.sortedBracketsWithDuplicates);
+      this.sortedBrackets.replaceAll((key, value) -> new ArrayList<>(value));
+      this.removeDuplicates(this.sortedBrackets);
+
+      PlayerInfo player = playerService.getCurrentPlayer();
+      LeaderboardRating ratingBean = player.getLeaderboardRatings().get(queue.getLeaderboard().technicalName());
+      double rating = (ratingBean != null) ? (ratingBean.mean() - 3 * ratingBean.deviation()) : 0;
+      this.playerBracketIndex = this.getPlayerBracketIndex(this.sortedBrackets, rating);
+
+      List<MapVersion> values = this.sortedBrackets.values().stream().flatMap(List::stream).toList();
+      this.resizeToContent(values.size(), TILE_SIZE);
+
+
+      fxApplicationThreadExecutor.execute(() -> values.forEach(this::addMapTile));
+    });
```

Comment:
A lot of this method sets values that are better off as properties so they can be more responsive and in line with the rest of the code base.

I would suggest taking a look at how some of the map methods are used by properties in other controllers.

This allows for greater flexibility in the  controller design so the developer doesn't have to make sure things are called in a specific order.

Additionally when already in a mono or flux pipeline rather than using fxapplicationexecutor.execute directly it is better to return the value and then use the publishOn method of the publisher.